packages/web/app/src/pages/target.tsx

```diff
@@ -209,41 +164,23 @@ function SchemaView(props: {
   organization: FragmentType<typeof SchemaView_OrganizationFragment>;
   project: FragmentType<typeof SchemaView_ProjectFragment>;
   target: FragmentType<typeof SchemaView_TargetFragment>;
-  highlightedService: string | null;
 }): ReactElement | null {
   const organization = useFragment(SchemaView_OrganizationFragment, props.organization);
   const project = useFragment(SchemaView_ProjectFragment, props.project);
   const target = useFragment(SchemaView_TargetFragment, props.target);
-  const [filterService, setFilterService] = useState(props.highlightedService ?? '');
-  const [term, setTerm] = useState(props.highlightedService ?? '');
-  const [openItems, setOpenItems] = useState<string[]>([]);
   const router = useRouter();
-  const debouncedFilter = useDebouncedCallback((value: string) => {
-    setFilterService(value);
-  }, 500);
-  const handleChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
-    event => {
-      const value = event.target.value;
-      debouncedFilter(value);
-      setFilterService(value);
-      setTerm(value);
-      setOpen(false);
-      setOpenItems(prevItems => [...new Set([...prevItems, value])]);
-      void router.navigate({
-        to: '/$organizationId/$projectId/$targetId',
-        search: { service: value },
-      });
-    },
-    [debouncedFilter, setTerm],
-  );
-  const reset = useCallback(() => {
-    setOpenItems([]);
-    setFilterService('');
-    setTerm('');
+  const selectedServiceName =
+    'service' in router.latestLocation.search &&
+    typeof router.latestLocation.search.service === 'string'
+      ? router.latestLocation.search.service
+      : null;
+
+  const [open, setOpen] = useState(false);
+  const reset = () => {
     void router.navigate({
-      to: '/$organizationId/$projectId/$targetId',
+      search: {},
     });
-  }, [setFilterService]);
```

Comment:
What `reset` now does is just updating the url, so our component re-renders and since we have no state, then there's no need to reset anything else.